src/composables/use-related.js

```diff
@@ -0,0 +1,25 @@
+import AudioService from '~/data/audio-service'
+import ImageService from '~/data/image-service'
+import { ref, useFetch } from '@nuxtjs/composition-api'
+import { AUDIO, IMAGE } from '~/constants/media'
+
+const services = { [AUDIO]: AudioService, [IMAGE]: ImageService }
+
+export default function useRelated({
+  mediaType,
+  mediaId,
+  service = services[mediaType],
+}) {
+  const media = ref([])
+  // fetch and fetchState are available as this.$fetch and this.$fetchState
+  // in components, so there's no need to export them
+  // eslint-disable-next-line no-unused-vars
```

Comment:
🤔 I think it would make more sense to still return the `fetchState`. Relying on the implicitly created `$fetchState` in the components is actually kind of confusing now that `useFetch` isn't even being called directly in the setup function. It might make a little more sense if `useRelated` was named `useFetchRelated`, because at least then where the `$fetchState` was coming from would be slightly more easy to investigate to a new-comer... but the easiest to understand would just be an explicit `fetchState` returned from the function and consumed in the component. That way there's absolutely 0 magic or obfuscation happening and someone reading the code and draw directly to the specific place the fetch state comes from.

Also... how would this work if there are multiple calls to `useFetch`? Do they just get magically merged into one magic `$fetchState` variable somehow? Or would that create a complication?

Reply:
My comment was based on the [Nuxt Composition API's useFetch documentation](https://composition-api.nuxtjs.org/lifecycle/usefetch/):

> $fetch and $fetchState will already be defined on the instance - so no need to return fetch or fetchState from setup.

I understand the confusion that this magic creates, but I also think it might be better to follow the advice of the library creators. Would adding a comment to the `useFetch` documentation and renaming the composable to useFetchRelated help?

Nuxt's fetch hooks already have this magic within them: If you have a fetch hook in your component, you can access the `$fetchState` object in the component, and it is probably not easy to understand where it comes from without knowing about `Nuxt`'s `fetch` hook. 
When I try returning `fetch` and `fetchState`, and using them in the component, they seem to work, but my editor complains...

Reply:
It's so strange that they would recommend relying on that magic! But if that's what the library authors think is best then I agree we can follow it. No need to rename it I guess, just an education thing!